File: Android/FitIn_v2/app/src/main/java/com/example/fitin_v2/ui/HomeActivity.java
package com.example.fitin_v2.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.os.Bundle;

import com.example.fitin_v2.R;
import com.example.fitin_v2.databinding.ActivityHomeBinding;
import com.example.fitin_v2.databinding.ActivityMainBinding;
import com.example.fitin_v2.ui.onboard.main.MainActivity;

public class HomeActivity extends AppCompatActivity {

    private ActivityHomeBinding binding;
    private HomeViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_home);

        viewModel = new ViewModelProvider(this).get(HomeViewModel.class);

        binding.setLifecycleOwner(this);
        binding.setHomeViewModel(viewModel);

        viewModel.onLogout().observe(this, logout -> {
            if (logout) {
                Intent intentMain = new Intent(getApplicationContext(), MainActivity.class);
                startActivity(intentMain);
                overridePendingTransition(0, 0);
                finish();
            }
        });

    }
}

File: Android/FitIn_Kotlin/app/src/main/java/com/example/fitin_kotlin/ui/home/HomeActivity.kt
package com.example.fitin_kotlin.ui.home

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.activity.viewModels
import androidx.databinding.DataBindingUtil
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.NavigationUI
import androidx.navigation.ui.setupWithNavController
import com.example.fitin_kotlin.R
import com.example.fitin_kotlin.databinding.ActivityHomeBinding
import com.example.fitin_kotlin.ui.bookmark.BookmarkViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.android.synthetic.main.activity_home.*

@AndroidEntryPoint
class HomeActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityHomeBinding = DataBindingUtil.setContentView(this,R.layout.activity_home)

        val navHostFragment = supportFragmentManager.findFragmentById(R.id.fcv_nav_home_fragment) as NavHostFragment
        val navController = navHostFragment.navController

        binding.bnvMenu.setupWithNavController(navController)

        navController.addOnDestinationChangedListener{ controller, destination, arguments ->
            when (destination.id) {
                R.id.newsDetailFragment -> bnv_menu.visibility = View.GONE
                R.id.recruitmentDetailFragment -> bnv_menu.visibility = View.GONE
                R.id.addBookmarkFragment -> bnv_menu.visibility = View.GONE
                R.id.createBookmarkFragment -> bnv_menu.visibility = View.GONE
                R.id.myNewsFragment -> bnv_menu.visibility = View.GONE
                R.id.myRecruitmentFragment -> bnv_menu.visibility = View.GONE
                R.id.bookmarkMyNewsFragment -> bnv_menu.visibility = View.GONE
                R.id.bookmarkMyRecruitmentFragment -> bnv_menu.visibility = View.GONE
                else -> bnv_menu.visibility = View.VISIBLE
            }
        }

    }
}